torch/distributed/_shard/sharded_tensor/_ops/linear.py

```diff
@@ -218,8 +215,17 @@ def _handle_col_wise_sharding(input, world_size, weight, rank, local_shard_t, bi
         result = torch.stack(results)  # type: ignore[arg-type]
     else:
         result = torch.cat(results)  # type: ignore[arg-type]
-    return _init_sharded_tensor_from_local_result(
-        weight, result, 0, -1, world_size, pg  # type: ignore[arg-type]
+    st_size = list(result.size())
+    st_size[-1] = weight.size(0)
+    new_sharding_spec = ChunkShardingSpec(
+        dim=-1,
+        placements=weight._sharding_spec.placements
```

Comment:
nit: lets use the public API: weight.sharding_spec() instead.

Reply:
sure.